packages/patternfly-4/react-core/src/layouts/Page/Page.js

```diff
@@ -16,25 +17,73 @@ const propTypes = {
   /** Header component (e.g. <PageHeader />) */
   header: PropTypes.node,
   /** Sidebar component for a side nav (e.g. <PageSidebar />) */
-  sidebar: PropTypes.node
+  sidebar: PropTypes.node,
+  /** condensed header on scroll */
+  useCondensed: PropTypes.bool,
+  /** condensed height override */
+  scrollingDistance: PropTypes.number
 };
 
 const defaultProps = {
   children: null,
   className: '',
   header: null,
-  sidebar: null
+  sidebar: null,
+  useCondensed: false,
+  scrollingDistance: 20
 };
 
-const Page = ({ className, children, header, sidebar, ...props }) => (
-  <div {...props} className={css(styles.page, className)}>
-    {header}
-    {sidebar}
-    <main role="main" className={css(styles.pageMain)}>
-      {children}
-    </main>
-  </div>
-);
+class Page extends React.Component {
+  state = { isCondensed: false };
+
+  constructor(props) {
+    super(props);
+    this.mainRef = React.createRef();
+  }
+  componentDidMount() {
+    const { useCondensed } = this.props;
+    if (useCondensed) {
+      // I picked this because it's approx 1 frame (ie: 16.7ms)
+      this.mainRef.current.addEventListener('scroll', debounce(this.handleScroll, 16));
+    }
+  }
+  componentWillUnmount() {
+    const { useCondensed } = this.props;
+    if (useCondensed) {
+      this.mainRef.current.removeEventListener('scroll', this.handleScroll);
+    }
+  }
+
+  handleScroll = e => {
+    window.requestAnimationFrame(() => {
```

Comment:
Should we always animate or should the consumer have the option to turn off animating?  Thinking about if they are on a slow network.

Reply:
right now.. you'd have to enable this feature to get the animation. But yes, love the idea of conditionally allowing animations as well as the feature itself. This kind of suggests some variation in CSS classes or the CSS styles themselves (thus the reason for considering runtime animation libraries of CSS).

Reply:
so coming back to this... after talking w/ @matthewcarleton today, i think the right course of action (at least for us) is to disable the scroll behavior on mobile and just display the condensed header. thoughts? Will try to confirm this tomorrow and update w/ a subsequent PR...